docs/source/python_api/mlflow.metrics.rst

```diff
@@ -86,14 +86,59 @@ We provide the following builtin factory functions to create :py:class:`Evaluati
 
 .. autofunction:: mlflow.metrics.rougeLsum
 
-.. autofunction:: mlflow.metrics.precision_at_k
-
 .. autofunction:: mlflow.metrics.toxicity
 
 .. autofunction:: mlflow.metrics.token_count
 
 .. autofunction:: mlflow.metrics.latency
 
+Retriever Metrics
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The following metrics are built-in metrics for the ``'retriever'`` model type, meaning they will be 
+automatically calculated with a default ``retriever_k`` value of 3. 
+
+To evaluate document retrieval models, it is recommended to use a dataset with the following 
+columns:
+
+- Input queries
+- Retrieved relevant doc IDs
+- Ground-truth doc IDs
+
+Alternatively, you can you can also provide a function through the ``model`` parameter to represent 
+your retrieval model. The function should take a Pandas DataFrame containing input queries and 
+ground-truth relevant doc IDs, and return a DataFrame with a column of retrieved relevant doc IDs.
+
+A "doc ID" is a string that uniquely identifies a document. All doc IDs should be entered as a 
+list of doc ID  strings.
+
+Parameters:
+
+- ``targets``: A string specifying the column name of the ground-truth relevant doc IDs
+- ``predictions``: A string specifying the column name of the retrieved relevant doc IDs in either 
+  the static dataset or the Dataframe returned by the ``model`` function
+- ``retriever_k``: A positive integer specifying the number of retrieved docs IDs to consider for 
+  each input query. ``retriever_k`` defaults to 3. You can change ``retriever_k`` by using the 
+  :py:func:`mlflow.evaluate` API:
+
+    1. ``evaluator_config={"retriever_k": 5}``
+    2. .. code-block:: python
+
+        extra_metrics = [
+            mlflow.metrics.precision_at_k(k=5),
+            mlflow.metrics.precision_at_k(6),
+            mlflow.metrics.recall_at_k(4),
+            mlflow.metrics.recall_at_k(5)
+        ]
```

Comment:
Suprised that this renders fine lol

Reply:
i was too, but I think this is the best way to handle multil-line python code in our docs